dependency/src/main/java/de/dagere/peass/execution/gradle/GradleBuildfileVisitor.java

```diff
@@ -121,35 +126,58 @@ public void visitMethodCallExpression(final MethodCallExpression call) {
       for (Expression argument : arguments.getExpressions()) {
          if (argument instanceof ClosureExpression) {
             ClosureExpression closure = (ClosureExpression) argument;
             BlockStatement blockStatement = (BlockStatement) closure.getCode();
             for (Statement statement : blockStatement.getStatements()) {
-               if (statement instanceof ExpressionStatement) {
-                  ExpressionStatement potentialSystemProperties = (ExpressionStatement) statement;
-                  if (potentialSystemProperties.getExpression() instanceof MethodCallExpression) {
-                     MethodCallExpression methodCallExpression = (MethodCallExpression) potentialSystemProperties.getExpression();
-
-                     if (methodCallExpression.getMethodAsString().equals("systemProperties")) {
-                        // TODO The following implementation would be better to assure usability with non-standard format (instead of manipulating string lines)
-                        // since I currently did not find a reliable way for reserialization, I leave it this way
-
-                        // ArgumentListExpression propertiesArguments = (ArgumentListExpression) methodCallExpression.getArguments();
-                        // MapExpression map = (MapExpression) propertiesArguments.getExpression(0);
-                        // for (MapEntryExpression expression : map.getMapEntryExpressions()) {
-                        // System.out.println(expression.getKeyExpression());
-                        // System.out.println(expression.getValueExpression());
-                        // }
-                        //
-                        // ConstantExpression keyExpression = new ConstantExpression("TEMP_DIR_PURE");
-                        // ConstantExpression valueExpression = new ConstantExpression("asdasd");
-                        // map.addMapEntryExpression(keyExpression, valueExpression);
-
-                        propertiesLine = methodCallExpression.getLineNumber() + offset;
+               ExpressionStatement potentialSystemProperties = null;
+               if (statement instanceof ReturnStatement) {
+                  Expression expression = ((ReturnStatement) statement).getExpression();
+                  if (expression.getText().contains("systemProperties") || expression.getText().contains("systemProperty")) {
+                     potentialSystemProperties = new ExpressionStatement(((ReturnStatement) statement).getExpression());
+                  }
+               } else if (statement instanceof ExpressionStatement) {
+                  potentialSystemProperties = (ExpressionStatement) statement;
+               }
+               if (potentialSystemProperties != null && potentialSystemProperties.getExpression() instanceof MethodCallExpression) {
+                  MethodCallExpression methodCallExpression = (MethodCallExpression) potentialSystemProperties.getExpression();
```

Comment:
This is (way) to big and should be extracted to a separate method. The rule of thumb is 10 lines are too much (which is not always considered, but should be in new code).

Reply:
This is refactored